Doc/whatsnew/3.10.rst

```diff
@@ -596,6 +602,54 @@ Changes in the Python API
   a 16-bit unsigned integer.
   (Contributed by Erlend E. Aasland in :issue:`42393`.)
 
+* The ``loop`` parameter to nearly all :mod:`asyncio` functions has been
+  removed following deprecation in Python 3.8.  Users should instead get the
+  current event loop inside of coroutines preferrably via
+  :func:`asyncio.get_running_loop` (Python 3.7 and above) or
+  :func:`asyncio.get_event_loop`.  For example, a coroutine that currently look
+  like this::
+
+     async def foo(loop):
+         ...
+
+  Can be replaced with this::
+
+     import asyncio
+
+     async def foo():
+         # Note that this asserts the loop is already running.
+         loop = asyncio.get_running_loop()
+         ...
```

Comment:
I don't see the relationship with the removal of the loop parameter. For example, loop parameter was removed from the sleep() function. We don't have to modifiy the sleep() implementation, only calls to this function.

My question on python-dev was if it's safe to replace sleep(loop=loop) with sleep().

Reply:
> I don't see the relationship with the removal of the loop parameter. For example, loop parameter was removed from the sleep() function. We don't have to modifiy the sleep() implementation, only calls to this function.

Yeah I realised my examples need touching up. Thanks for pointing that out.

> My question on python-dev was if it's safe to replace sleep(loop=loop) with sleep().

Disclaimer: I'm not an asyncio internals expert by any means, so please take my opinion with a grain of salt.

Depends on the user code. IMO, for the most part, yes. But there are certain wacky scenarios where it's not. Asyncio has some checks against passing a loop to a coroutine that isn't executing in that loop:

So obviously wrong stuff like (in python 3.9)
```python
>>> import asyncio
>>> loop1 = asyncio.new_event_loop()
>>> loop2 = asyncio.new_event_loop()
>>> loop1.run_until_complete(asyncio.sleep(1, loop=loop2))
RuntimeError: Task <Task pending ... got Future attached to a different loop
```
However, I have no clue how creative some people's code can be ;).

Reply:
I don't think that asyncio supports two event loops running at the same time. I guess that the common case is a single loop. It's just that previously, asyncio strongly adviced to always pass loop explicitly. Now it's the opposite: passing a loop raises an error...

Reply:
I'm going to wait for one of the experts on the bpo to give judgement on this, because I don't feel comfortable giving an opinion on this.

Reply:
@vstinner 
> I don't think that asyncio supports two event loops running at the same time. 

Asyncio supports multiple running event loops at the same time as long as they are in their own separate threads, just not in the same thread like in @Fidget-Spinner's example. E.g.

```py
>>> import asyncio

>>> import asyncio
>>> async def show_run(name, count):
	loop = asyncio.get_running_loop()
	for i in range(count):
		await asyncio.sleep(2)
		print(f"{name} loop: {loop}")
		
>>> def loop_other_thread(name, count):
	print(f"start {name}")
	loop = asyncio.new_event_loop()
	asyncio.set_event_loop(loop)
	loop.run_until_complete(show_run(name, count))
	loop.close()
	print(f"end {name}")

>>> t1 = threading.Thread(target=loop_other_thread, args=('t1', 5))
>>> t2 = threading.Thread(target=loop_other_thread, args=('t2', 3))
>>> t1.start()
start t1
t1 loop: <_WindowsSelectorEventLoop running=True closed=False debug=False>
>>> t2.start()
start t2
>>> 
t1 loop: <_WindowsSelectorEventLoop running=True closed=False debug=False>
t2 loop: <_WindowsSelectorEventLoop running=True closed=False debug=False>
t1 loop: <_WindowsSelectorEventLoop running=True closed=False debug=False>
t2 loop: <_WindowsSelectorEventLoop running=True closed=False debug=False>
t1 loop: <_WindowsSelectorEventLoop running=True closed=False debug=False>
t2 loop: <_WindowsSelectorEventLoop running=True closed=False debug=False>
end t2
t1 loop: <_WindowsSelectorEventLoop running=True closed=False debug=False>
end t1
```

To make use of this in a practical scenario, one would presumably have access to both loops from the main thread and send coroutines to each via run_coroutine_threadsafe(). Although it's definitely less common, I think it's worth addressing via [my suggestion below](https://github.com/python/cpython/pull/24256#discussion_r560511567)